solr/core/src/java/org/apache/solr/update/AddUpdateCommand.java

```diff
@@ -96,12 +96,21 @@ public SolrInputDocument getSolrInputDocument() {
    * Any changes made to the returned Document will not be reflected in the SolrInputDocument, or future calls to this
    * method.
    * Note that the behavior of this is sensitive to {@link #isInPlaceUpdate()}.
+   * @param withBlockId If true, then block id is forcibly added to the doc
    */
-   public Document getLuceneDocument() {
+   Document getLuceneDocument(boolean withBlockId) {
```

Comment:
can we call this addRootField ?

Reply:
actually I'm not sure we actually need to overload getLuceneDocument with this boolean after all.  if "isInPlaceUpdate()" then don't add root but otherwise add it.  That'll be fine; right?